packages/firestore/android/src/reactnative/java/io/invertase/firebase/firestore/ReactNativeFirebaseFirestoreCollectionModule.java

```diff
@@ -193,6 +198,99 @@ public void collectionCount(
             });
   }
 
+  @ReactMethod
+  public void aggregateQuery(
+    String appName,
+    String databaseId,
+    String path,
+    String type,
+    ReadableArray filters,
+    ReadableArray orders,
+    ReadableMap options,
+    ReadableArray aggregateQueries,
+    Promise promise
+  ){
+    FirebaseFirestore firebaseFirestore = getFirestoreForApp(appName, databaseId);
+    ReactNativeFirebaseFirestoreQuery firestoreQuery =
+      new ReactNativeFirebaseFirestoreQuery(
+        appName,
+        databaseId,
+        getQueryForFirestore(firebaseFirestore, path, type),
+        filters,
+        orders,
+        options);
+
+    ArrayList<AggregateField> aggregateFields = new ArrayList<>();
+
+    for (int i = 0; i < aggregateQueries.size(); i++) {
+      ReadableMap aggregateQuery = aggregateQueries.getMap(i);
+
+      String aggregateType = aggregateQuery.getString("aggregateType");
+      String fieldPath = aggregateQuery.getString("field");
+
+      assert aggregateType != null;
+        switch (aggregateType) {
+          case "count":
+            aggregateFields.add(AggregateField.count());
+            break;
+          case "sum":
+            assert fieldPath != null;
+            aggregateFields.add(AggregateField.sum(fieldPath));
+            break;
+          case "average":
+            assert fieldPath != null;
+            aggregateFields.add(AggregateField.average(fieldPath));
+            break;
+          default:
+            throw new Error("Invalid AggregateType: " + aggregateType);
+        }
+    }
+      AggregateQuery firestoreAggregateQuery = firestoreQuery.query.aggregate(aggregateFields.get(0),
+      aggregateFields.subList(1, aggregateFields.size()).toArray(new AggregateField[0]));
+
+      firestoreAggregateQuery
+        .get(AggregateSource.SERVER)
+        .addOnCompleteListener(
+          task -> {
+            if (task.isSuccessful()) {
+              WritableMap result = Arguments.createMap();
+              AggregateQuerySnapshot snapshot = task.getResult();
+
+              for (int k = 0; k < aggregateQueries.size(); k++) {
+                ReadableMap aggQuery = aggregateQueries.getMap(k);
+                String aggType = aggQuery.getString("aggregateType");
+                String field = aggQuery.getString("field");
+                String key = aggQuery.getString("key");
+                assert key != null;
+                assert aggType != null;
```

Comment:
Are any of these asserts useful? It was my understanding that assert verifications are disabled by default so this would never really trigger anything, and if it did trigger the whole app dies (via an AssertionError) without notice of what did it since there is no exception handler set to catch Error (vs a try/catch with typical catch of Exception wrapping a block where you use Object.requireNonNull

Reply:
Removed in place of `Object.requireNonNull`. Largely to remove warnings in android studio 👍